pkg/manager/member/tiflash_upgrader.go

```diff
@@ -101,6 +110,17 @@ func (u *tiflashUpgrader) Upgrade(tc *v1alpha1.TidbCluster, oldSet *apps.Statefu
 			if store.State != v1alpha1.TiKVStateUp {
 				return controller.RequeueErrorf("tidbcluster: [%s/%s]'s upgraded TiFlash pod: [%s], store state is not UP", ns, tcName, podName)
 			}
+			if ver, err := semver.NewVersion(tc.TiFlashVersion()); err == nil && ver.Compare(tiflashV512) >= 0 {
+				status, err := u.deps.TiFlashControl.GetTiFlashPodClient(tc.Namespace, tc.Name, podName, tc.IsTLSClusterEnabled()).GetStoreStatus()
+				if err != nil {
+					return controller.RequeueErrorf("tidbcluster: [%s/%s]'s upgraded TiFlash pod: [%s], get store status failed: %s", ns, tcName, podName, err)
+				}
+
+				if status != tiflashapi.Running {
```

Comment:
This works for TiFlash with only 1 replica, right?

Reply:
In my understanding, it doesn't enter this loop. Does the function Upgrade return in L73 after the last Pod upgraded?

Reply:
It's possible that it enters this loop when the status of statefulset is not updated yet.

Reply:
make sense, i think it also work

Reply:
Yes, we need to test this case and make sure it works.